File: ecommerce-java/user-service/src/main/java/com/ecommerce/userservice/dto/ResponseUser.java
package com.ecommerce.userservice.dto;

import com.ecommerce.userservice.domain.UserEntity;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ResponseUser {
    private String email;
    private String name;
    private String userId;

    private List<ResponseOrder> orders;


    public static ResponseUser userEntityToResponseUser(UserEntity userEntity){
        return new ResponseUser(
                userEntity.getEmail(),
                userEntity.getName(),
                userEntity.getUserId(),
                new ArrayList<>()
        );
    }
}

File: ecommerce-kotlin/user-service/src/main/kotlin/com/example/userservice/dto/ResponseUser.kt
package com.example.userservice.dto

import com.example.userservice.domain.UserEntity

data class ResponseUser(
    val email: String,
    val name: String,
    val userId: String,
    val orders: List<ResponseOrder>
) {
    constructor(userEntity: UserEntity):this(
        userEntity.email, userEntity.name, userEntity.userId, arrayListOf()
    )
}
